pkg/os/shell/shell_windows.go

```diff
@@ -43,21 +43,22 @@ func getProcessEntry(pid int) (pe *syscall.ProcessEntry32, err error) {
 }
 
 // getNameAndItsPpid returns the exe file name its parent process id.
-func getNameAndItsPpid(pid int) (exefile string, parentid int, err error) {
+func getNameAndItsPpid(pid uint32) (exefile string, parentid uint32, err error) {
 	pe, err := getProcessEntry(pid)
 	if err != nil {
 		return "", 0, err
 	}
 
 	name := syscall.UTF16ToString(pe.ExeFile[:])
-	return name, int(pe.ParentProcessID), nil
+	return name, pe.ParentProcessID, nil
 }
 
 func detect() (string, error) {
 	shell := os.Getenv("SHELL")
 
 	if shell == "" {
-		shell, shellppid, err := getNameAndItsPpid(os.Getppid())
+		//#nosec G115 -- pid is coming from os.Getppid() as int
```

Comment:
Why is this ok to ignore the overflow?
I'd expect some range check before the cast
```
if pid < 0 || pid > math.MaxUint32 {
  // error
}
```

Reply:
Do you think `oc.Getppid` return something out of range of int? Shouldn't than it would be checked on os module  level?

Reply:
If these are the assumptions you want to make to ignore this warning, add them to the `nosec` comment. Imo it does not hurt to have checks to make it very clear.

Reply:
Added range check